### DISCUSSION TITLE
Testing locally, linking to another repository

### DISCUSSION THREAD

**Discussion by rgplantz** [community/community#23400] [Other Feature Feedback, Questions, & Ideas]
<p>In my <a href="https://rgplantz.github.io" rel="noopener nofollow ugc">rgplantz.github.io</a> I have an <code>index.md</code> file that links to another repository with <code>[My example](./example/)</code>. I also have an <code>index.md</code> file in the <code>gh-pages</code> branch of the <code>example</code> repository. It works quite nicely online.</p>
<p>I would like to test things locally. I cloned both repositories to my computer. From within the <code>rgplantz.github.io</code> local directory I use <code>bundle exec jekyll serve</code>. The content of the <code>index.md</code> file in <code>rgplantz.github.io</code> renders very nicely at <code>localhost:4000</code>. But when I click on the <code>My example</code> link, I get a 404 error.</p>
<p>Is there any way to get this link to work locally like it does on the GitHub server?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>If it’s a separate repository, rather than just a directory inside the same repository, then there isn’t a way to make this work locally too, no.</p>
<p>To get it to work locally you’d need to clone the second repository and host it using <code>bundle exec jekyll serve</code>, but it would also need to be served on the same port <code>4000</code>, which isn’t possible if you’re running 2 separate webservers to host both repos locally.</p>

**Comment by rgplantz** [comment]
<p>Thank you, thomasshaped. Pretty much as I had guessed.</p>
<p>This is a site for two books. I’m providing both answers (text and figures) to exercises and code source. I originally thought that a good organization would be to place the code source files in the main branch and the answers in the gh-pages branch. Now I’m thinking that a better organization is to place the answers in separate directories in the primary pages repository (<a href="http://rgplantz.github.io" rel="noopener nofollow ugc">rgplantz.github.io</a>) and the code sources in separate repositories. That would easier for me to work locally on the answers text.</p>
